Dealer.h

```diff
@@ -0,0 +1,47 @@
+#ifndef DEALER_H
+#define DEALER_H
+#include "Deck.h"
+#include <iostream>
+#include <vector>
+#include <string>
+#include <string_view>
+class Dealer
+{
+private:
+  Deck deck{};
+  std::string m_name{};
+  std::vector<std::string> Hand{};
+  int m_score{};
+  static const int c_maxScore{21};
+public:
+  Dealer(std::string_view name)
+    : m_name{name}
+  {
+  }
+
+  friend std::ostream& operator<<(std::ostream& out, const Dealer& d)
+  {
+    out << "Dealer " << d.m_name << " has a score of << " << d.m_score << "\nHas cards: \n";
+    int it{};
+    for(const auto& cardStrings : d.Hand)
```

Comment:
Well I have not actually gotten the program to the point that its behavior is something I can check. But my intent with this loop was to print each card held by the dealer, which can be accomplished by the ranged for loop without calling the .at(it). I believe behavior should be right, and changing it so its not a weird ranged for loop hybrid, as I show, should hopefully capture that behavior.

Reply:
I'm not sure if the behavior will be quite right as it is, but getting rid of the `.at()` call should make it right I think. The `.at()` is just going to print a single character of the string at the index you give it and it sounds like you want to print the whole string, right?

Reply:
`    switch(rankForComparison) 
    {
        case(Card::Rank::two)     :    card_string = "2"  ;       break;
        case(Card::Rank::three)   :    card_string = "3"  ;       break;           
        case(Card::Rank::four)    :    card_string = "4"  ;       break;
        case(Card::Rank::five)    :    card_string = "5"  ;       break;
        case(Card::Rank::six)     :    card_string = "6"  ;       break;
        case(Card::Rank::seven)   :    card_string = "7"  ;       break;
        case(Card::Rank::eight)   :    card_string = "8"  ;       break;
        case(Card::Rank::nine)    :    card_string = "9"  ;       break;
        case(Card::Rank::ten)     :    card_string = "T"  ;       break;
        case(Card::Rank::jack)    :    card_string = "J"  ;       break;
        case(Card::Rank::queen)   :    card_string = "Q"  ;       break;
        case(Card::Rank::king)    :    card_string = "K"  ;       break;
        case(Card::Rank::ace)     :    card_string = "A"  ;       break;
    }

    switch(suitForComparison) 
    {
        case (Card::Suit::spade)  :    card_string += "S" ;         break;
        case (Card::Suit::club)   :    card_string += "C" ;         break;
        case (Card::Suit::diamond):    card_string += "D" ;         break;
        case (Card::Suit::heart)  :    card_string += "H" ;         break;
    }`

(I wonder how that will format here).

I"m attempting to have the print overloads for Players and Dealers is print and empty-space separate each card, that right now in Deck.cpp above, is formatted like "3C 6S QH". Each player would have a different sized vector<string> Hand, depending on how many times player hit and then stayed. That will be true of dealer also.

I'm really just messing around with this little program, but I think it would be cool to get the number of players and names to be user inputs. And eventually use this whole program in a server/client with socket functions